CMakeLists.txt

```diff
@@ -228,8 +226,14 @@ if(ENABLE_MANUAL)
     message(WARNING "Found no *nroff program")
   endif()
 endif()
-# Required for building manual, docs, tests
-find_package(Perl REQUIRED)
+
+option(ENABLE_DOCS "to build docs" ON)
+option(BUILD_TESTING "to build tests" ON)
+
+if(ENABLE_MANUAL OR ENABLE_DOCS OR BUILD_TESTING)
+   # Required for building manual, docs, tests
+   find_package(Perl REQUIRED)
+endif()
```

Comment:
This is the perl related section already being discussed and work on in #1600. Please avoid that in your PR.